modules/sdk-coin-atom/src/lib/keyPair.ts

```diff
@@ -0,0 +1,74 @@
+import { randomBytes } from 'crypto';
+import { bip32 } from '@bitgo/utxo-lib';
+import { pubkeyToAddress } from '@cosmjs/amino';
+import { toBase64 } from '@cosmjs/encoding';
+import {
+  DefaultKeys,
+  isPrivateKey,
+  isPublicKey,
+  isSeed,
+  KeyPairOptions,
+  Secp256k1ExtendedKeyPair,
+} from '@bitgo/sdk-core';
+
+const DEFAULT_SEED_SIZE_BYTES = 16;
+
+/**
+ * Cosmos keys and address management.
+ */
+export class KeyPair extends Secp256k1ExtendedKeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key (extended or raw), or a public key
+   *     (extended, compressed, or uncompressed)
+   */
+  constructor(source?: KeyPairOptions) {
+    super(source);
+    if (!source) {
+      const seed = randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      this.hdNode = bip32.fromSeed(seed);
+    } else if (isSeed(source)) {
+      this.hdNode = bip32.fromSeed(source.seed);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+
+    if (this.hdNode) {
+      this.keyPair = Secp256k1ExtendedKeyPair.toKeyPair(this.hdNode);
+    }
+  }
+
+  /**
+   * Cosmos default keys format: raw private and compressed public key
+   *
+   * @returns { DefaultKeys } The keys in the protocol default key format
+   */
+  getKeys(): DefaultKeys {
+    return {
+      pub: this.getPublicKey({ compressed: true }).toString('hex'),
```

Comment:
> A compressed format is used for secp256k1 and secp256r1 serialization.
https://docs.cosmos.network/v0.46/basics/accounts.html

Need to double check if this is what is wanted

Reply:
While I was trying to create a wallet address using the `@cosmjs/amino` library, it complained that I was having an `Invalid Secp256k1 pubkey length (compressed): 48` and that the correct compressed length was 33 bytes - [Link 1](https://github.com/cosmos/cosmjs/blob/main/packages/amino/src/pubkeys.ts#L42-L53), [Link 2](https://github.com/cosmos/cosmjs/blob/main/packages/amino/src/addresses.ts#L17). From this I believe compressed is what's expected but I'll defer to another pair of eyes.